test/common/stats/thread_local_store_test.cc

```diff
@@ -80,6 +80,77 @@ class StatsThreadLocalStoreTest : public testing::Test, public RawStatDataAlloca
     store_->addSink(sink_);
   }
 
+  std::vector<uint64_t> h1_cumulative_values, h2_cumulative_values, h1_interval_values,
+      h2_interval_values;
+
+  int validateMerge() {
+
+    std::shared_ptr<std::atomic<bool>> merge_called = std::make_shared<std::atomic<bool>>(false);
+    store_->mergeHistograms([merge_called]() -> void { *merge_called = true; });
+
+    EXPECT_TRUE(*merge_called);
+
+    std::list<HistogramSharedPtr> histogram_list = store_->histograms();
+
+    histogram_t* hist1_cumulative = hist_alloc();
+    for (uint64_t value : h1_cumulative_values) {
+      hist_insert_intscale(hist1_cumulative, value, 0, 1);
+    }
+
+    histogram_t* hist2_cumulative = hist_alloc();
+    for (uint64_t value : h2_cumulative_values) {
+      hist_insert_intscale(hist2_cumulative, value, 0, 1);
+    }
+
+    histogram_t* hist1_interval = hist_alloc();
+    for (uint64_t value : h1_interval_values) {
+      hist_insert_intscale(hist1_interval, value, 0, 1);
+    }
+
+    histogram_t* hist2_interval = hist_alloc();
+    for (uint64_t value : h2_interval_values) {
+      hist_insert_intscale(hist2_interval, value, 0, 1);
+    }
+
+    HistogramStatisticsImpl h1_cumulative_statistics(hist1_cumulative);
+    HistogramStatisticsImpl h2_cumulative_statistics(hist2_cumulative);
+    HistogramStatisticsImpl h1_interval_statistics(hist1_interval);
+    HistogramStatisticsImpl h2_interval_statistics(hist2_interval);
+
+    for (const Stats::HistogramSharedPtr& histogram : histogram_list) {
+      if (histogram->name().find("h1") != std::string::npos) {
+        EXPECT_EQ(histogram->cumulativeStatistics().summary(), h1_cumulative_statistics.summary());
+        EXPECT_EQ(histogram->intervalStatistics().summary(), h1_interval_statistics.summary());
+      } else {
+        EXPECT_EQ(histogram->cumulativeStatistics().summary(), h2_cumulative_statistics.summary());
+        EXPECT_EQ(histogram->intervalStatistics().summary(), h2_interval_statistics.summary());
+      }
+    }
+
+    hist_free(hist1_cumulative);
+    hist_free(hist2_cumulative);
+    hist_free(hist1_interval);
+    hist_free(hist2_interval);
+
+    h1_interval_values.clear();
+    h2_interval_values.clear();
+
+    return histogram_list.size();
+  }
+
+  void expectCallAndAccumulate(Histogram& histogram, uint64_t record_value) {
+    EXPECT_CALL(sink_, onHistogramComplete(Ref(histogram), record_value));
+    histogram.recordValue(record_value);
+
+    if (histogram.name().find("h1") != std::string::npos) {
```

Comment:
can you comment what you expect histogram name to look like here?  If this is not exactly "h1" could you maybe use absl::StartsWith rather than 'find'?

Reply:
ping